src/telemetry/ExperimentationTelemetry.ts

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { IExperimentationTelemetry } from 'vscode-tas-client';
+
+export class ExperimentationTelemetry implements IExperimentationTelemetry {
+    private readonly sharedProperties: { [key: string]: string } = {};
+
+    /**
+     * "Handles" telemetry events by adding the shared properties to them
+     * @param context The action context
+     */
+    public async handleTelemetry(context: IActionContext): Promise<void> {
+        // Copy shared properties into event properties
+        for (const key of Object.keys(this.sharedProperties)) {
+            if (context.telemetry.properties[key]) {
+                console.error('Local telemetry property will be overwritten by shared property.');
+            }
+
+            context.telemetry.properties[key] = this.sharedProperties[key];
+        }
+    }
+
+    /**
+     * Implements `postEvent` for `IExperimentationTelemetry`. This implementation is admittedly strange, but necessary since `ITelemetryReporter` is no longer public.
+     * @param eventName The name of the event
+     * @param props The properties to attach to the event
+     */
+    public postEvent(eventName: string, props: Map<string, string>): void {
+        // This implementation is admittedly strange, but necessary since ITelemetryReporter is no longer public
```

Comment:
6 months from now no one will remember what `ITelemetryReporter` was. I would just drop the "this is strange" part.